agent/consul/acl_endpoint.go

```diff
@@ -1405,8 +1470,653 @@ func (a *ACL) RoleResolve(args *structs.ACLRoleBatchGetRequest, reply *structs.A
 			return acl.ErrPermissionDenied
 		}
 	}
+	for _, roleName := range args.RoleNames {
+		if role, ok := nameMap[roleName]; ok {
+			// only add non-deleted roles
+			if role != nil {
+				reply.Roles = append(reply.Roles, role)
+			}
+		} else {
+			// send a permission denied to indicate that the request included
+			// role names not associated with this token
+			return acl.ErrPermissionDenied
+		}
+	}
 
 	a.srv.setQueryMeta(&reply.QueryMeta)
 
 	return nil
 }
+
+func (a *ACL) RoleBindingRuleRead(args *structs.ACLRoleBindingRuleGetRequest, reply *structs.ACLRoleBindingRuleResponse) error {
+	if err := a.aclPreCheck(); err != nil {
+		return err
+	}
+
+	if done, err := a.srv.forward("ACL.RoleBindingRuleRead", args, args, reply); done {
+		return err
+	}
+
+	if rule, err := a.srv.ResolveToken(args.Token); err != nil {
+		return err
+	} else if rule == nil || !rule.ACLRead() {
+		return acl.ErrPermissionDenied
+	}
+
+	return a.srv.blockingQuery(&args.QueryOptions, &reply.QueryMeta,
+		func(ws memdb.WatchSet, state *state.Store) error {
+			index, rule, err := state.ACLRoleBindingRuleGetByID(ws, args.RoleBindingRuleID)
+
+			if err != nil {
+				return err
+			}
+
+			reply.Index, reply.RoleBindingRule = index, rule
+			return nil
+		})
+}
+
+// TODO (DEPRECATE?)
```

Comment:
Nuking this as this method endpoint was only used for replication in Policies and Roles and so it's not needed for role binding rules since they aren't replicated.